src/wallet/test/util.cpp

```diff
@@ -61,6 +61,27 @@ CTxDestination getNewDestination(CWallet& w, OutputType output_type)
     return *Assert(w.GetNewDestination(output_type, ""));
 }
 
+MockableCursor::MockableCursor(const std::map<SerializeData, SerializeData>& records, bool pass, Span<std::byte> prefix)
+{
+    m_pass = pass;
+
+    // Start the cursor at the first value that is greater than or equal to the prefix
+    m_cursor = records.lower_bound(SerializeData(prefix.begin(), prefix.end()));
+
+    // The end cursor is the first item that is greater than or equal to the prefix + 1 (when interpreted as an integer)
+    SerializeData end_range(prefix.begin(), prefix.end());
+    auto it = end_range.rbegin();
+    for (; it != end_range.rend(); ++it) {
+        if (*it == std::byte(0xff)) {
+            *it = std::byte(0);
+            continue;
+        }
+        *it = std::byte(std::to_integer<unsigned char>(*it) + 1);
+        break;
+    }
+    m_cursor_end = records.lower_bound(end_range);
```

Comment:
In commit "wallet: Add GetPrefixCursor to DatabaseBatch" (42cce1d43be0e662783abf4af60283c352297eac)

I think you can just do:

```c++
m_cursor_end = records.upper_bound(SerializeData(prefix.begin(), prefix.end()));
```

or

```c++
std::tie(m_cursor, m_cursor_end) = records.equal_range(SerializeData(prefix.begin(), prefix.end()));
```

And don't need the for loop.

Reply:
I don't think `upper_bound` or `equal_range` will work. They will find the upper bound to be the first item that is greater than our prefix, but any record with the right prefix and is longer than the prefix will be greater than it, so we end up actually only getting the record(s) that exactly match the prefix.

Reply:
Adopted these suggestions.